app/models/service_template_terraform_template.rb

```diff
@@ -41,4 +44,23 @@ def terraform_template(action)
 
     ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Template.find(template_id)
   end
+
+  def create_dialogs(config_info)
+    # [:provision, :retirement, :reconfigure].each_with_object({}) do |action, hash|
+    [:provision,].each_with_object({}) do |action, hash|
```

Comment:
No sense in looping over a single key, also best not to commit commented out code

Reply:
Though still using the loop, as latter we will have`:retirement, :reconfiguration`.

But then you are right, we may not need `:retirement, :reconfiguration`,   as currently use case for terraform is to use the same template,  with same dialog for both `:retirement, :reconfiguration`.

Reply:
Made the changes to create new dialog for :provision action,  and use the same dialog_id with both :retirement and :reconfiguration action.